File: Backend/eCommerce.Application/DTOs/Wishlist/WishlistModels.cs
namespace eCommerce.Application.DTOs.Wishlist
{
    public class AddToWishlistRequest
    {
        public Guid ProductId { get; set; }
        public string UserId { get; set; } = string.Empty;
    }

    public class WishlistResponse
    {
        public List<WishlistItem> Items { get; set; } = new();
        public int TotalCount { get; set; }
    }

    public class WishlistItem
    {
        public Guid Id { get; set; }
        public Guid ProductId { get; set; }
        public string ProductName { get; set; } = string.Empty;
        public string ProductDescription { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public string ImageUrl { get; set; } = string.Empty;
        public string CategoryName { get; set; } = string.Empty;
        public bool IsInStock { get; set; }
        public DateTime AddedAt { get; set; }
    }
}

File: Frontend/src/app/Core/Models/wishlist.models.ts
export interface AddToWishlistRequest {
  productId: string;
  userId: string;
}

export interface WishlistResponse {
  items: WishlistItem[];
  totalCount: number;
}

export interface WishlistItem {
  id: string;
  productId: string;
  productName: string;
  productDescription: string;
  price: number;
  imageUrl: string;
  categoryName: string;
  isInStock: boolean;
  addedAt: string;
}